### ISSUE TITLE
add linuguistics

### ISSUE THREAD

**Issue by felixbur** [felixbur/nkululeko#94]
Nkululeko could be multimodal, if a 
transcript
field is added to the audio files
and then, 
linguistic features extractors could be added to the feature_sets

**Comment by bagustris** [comment]
Some datasets already have transcriptions (but I skip that since I don't think it will be needed). It can be added as an additional column in the CSV or audformat. If there is no transcription, we can utilize hugging face (such as a whisper) to generate transcripts during pre-processing in each dataset. Then, the "linguistic feature extractor" will process transcription in the `transcript` column (I propose this name as the header of transcription) to generate _word embeddings_ (linguistic feature).

This is useful to use speech along with transcription for the detection of such degradation like Alzheimer's.

**Comment by bagustris** [comment]
If there is a transcript field/column in the CSV file (assuming format is CSV), this can be easily solved using speechbrain package.

https://speechbrain.readthedocs.io/en/latest/API/speechbrain.wordemb.transformer.html

e.g., 

file| label | transcript
----|---|----
1.wav | happy | "I am happy"

**Comment by felixbur** [comment]
to me it seems most intuitive to extend the *predict* module by 
* text transcript
* speaker id
thereby solving [#167](https://github.com/felixbur/nkululeko/issues/167)

**Comment by bagustris** [comment]
Based on my previous paper [1], GloVe embedding is the most powerful one for bimodal emotion recognition task; hence, I would like tackle this issue with GloVe embedding.

[1] Atmaja, B. T., & Akagi, M. (2020). Improving Valence Prediction in Dimensional Speech Emotion Recognition Using Linguistic Information. Proceedings of 2020 23rd Conference of the Oriental COCOSDA International Committee for the Co-Ordination and Standardisation of Speech Databases and Assessment Techniques, O-COCOSDA 2020, 166–171. https://doi.org/10.1109/O-COCOSDA50338.2020.9295032

**Comment by bagustris** [comment]
Yes, how to tackle this? should we add new values for [FEAT] section?

````
[FEATS] 
linguistic = 'glove' 
language = en
```

We need to think how to make it modular so we can switch from one model (glove) to other models (bert, fastext, etc).

**Comment by felixbur** [comment]
ok, i added an issue https://github.com/felixbur/nkululeko/issues/213 to add whisper for speech2text and will do this now

**Comment by bagustris** [comment]
@felixbur 

I thinks this issues are already solved, right?

One of my students utilized BERT from German and Japanese, and it seems to work in addition to English BERT.
